src/main/java/com/minecolonies/coremod/entity/ai/citizen/cook/EntityAIWorkCook.java

```diff
@@ -0,0 +1,561 @@
+     */
+    private static final int INTELLIGENCE_MULTIPLIER = 1;
+
+    /**
+     * This times the building level the worker has to keep.
+     */
+    private static final int LEAST_KEEP_FOOD_MULTIPLIER = 10;
+
+    /**
+     * The amount of food which should be served to the woker.
+     */
+    private static final int AMOUNT_OF_FOOD_TO_SERVE = 3;
+
+    /**
+     * Delay between each serving.
+     */
+    private static final int SERVE_DELAY = 30;
+
+    /**
+     * The standard delay after each terminated action.
+     */
+    private static final int STANDARD_DELAY = 5;
+
+    /**
+     * Predicate describing food.
+     */
+    private static final Predicate<ItemStack> isFood = itemStack -> !ItemStackUtils.isEmpty(itemStack) && itemStack.getItem() instanceof ItemFood;
+
+    /**
+     * Predicate describing a cookable itemStack.
+     */
+    private static final Predicate<ItemStack> isCookable = isFood.and(itemStack -> !ItemStackUtils.isEmpty(FurnaceRecipes.instance().getSmeltingResult(itemStack)));
+
+    /**
+     * The citizen the worker is currently trying to serve.
+     */
+    private final List<EntityCitizen> citizenToServe = new ArrayList<>();
+
+    /**
+     * The warehouse he found.
+     */
+    private BuildingWareHouse wareHouse = null;
+
+    /**
+     * The current position the worker should walk to.
+     */
+    private BlockPos walkTo = null;
+
+    /**
+     * The building range the cook should search for clients.
+     */
+    private AxisAlignedBB range = null;
+
+    /**
+     * Constructor for the Cook.
+     * Defines the tasks the cook executes.
+     *
+     * @param job a cook job to use.
+     */
+    public EntityAIWorkCook(@NotNull final JobCook job)
+    {
+        super(job);
+        super.registerTargets(
+                new AITarget(IDLE, START_WORKING),
+                new AITarget(START_WORKING, this::startWorking),
+                new AITarget(COOK_GET_FOOD, this::gatherFoodFromWarehouse),
+                new AITarget(COOK_GATHERING, this::gatherFoodFromBuilding),
+                new AITarget(COOK_COOK_FOOD, this::cookFood),
+                new AITarget(COOK_SERVE, this::serveFood),
+                new AITarget(COOK_RETRIEVE_FOOD, this::retrieve),
+                new AITarget(COOK_GET_FIREWOOD, this::getBurnableMaterial)
+        );
+        worker.setSkillModifier(CHARISMA_MULTIPLIER * worker.getCitizenData().getCharisma()
+                + INTELLIGENCE_MULTIPLIER * worker.getCitizenData().getIntelligence());
+        worker.setCanPickUpLoot(true);
+    }
+
+    private AIState getBurnableMaterial()
+    {
+        if(walkTo == null && walkToBuilding())
+        {
+            return getState();
+        }
+
+        if (getOwnBuilding().getCountOfPredicateInHut(TileEntityFurnace::isItemFuel, 1, world) < 1)
+        {
+            checkOrRequestItemsAsynch(false, new ItemStack(Blocks.LOG), new ItemStack(Blocks.LOG2), new ItemStack(Items.COAL));
+            setDelay(400);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 400 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)